### PR TITLE
fix(TCOMP-2277): Complete the TCK schema's entry properties

### PR DESCRIPTION
**Pull request by wwang-talend**
### Requirements

* Any code change adding any logic **MUST** be tested through a unit test executed with the default build
* Any API addition **MUST** be done with a documentation update if relevant 

### Why this PR is needed?
https://jira.talendforge.org/browse/TCOMP-2277

### What does this PR adds (design/code thoughts)?
https://jira.talendforge.org/browse/TCOMP-2277

### CODE REVIEW

File: component-api/src/main/java/org/talend/sdk/component/api/record/EntryProperty.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.talend.sdk.component.api.record;
+
+public interface EntryProperty {
```
Comment (undx) [inline component-api/src/main/java/org/talend/sdk/component/api/record/EntryProperty.java:18]: In framework, I prefer `SchemaProperty` as naming for this class as we may also add some props at schema  level and not just enties.

**Review by undx — COMMENTED** [review]
I think that we also need to unite keys in component-runtime-di...

File: component-api/src/main/java/org/talend/sdk/component/api/record/EntryProperty.java (hunk 1 above)
Comment (wwang-talend) [reply component-api/src/main/java/org/talend/sdk/component/api/record/EntryProperty.java:18]: Hi @undx 
about "unite keys in component-runtime-di", that may need migration, will do it in TDI-48755 after TDI-48634 pass Pipeline team's usage.

Comment (wwang-talend) [reply component-api/src/main/java/org/talend/sdk/component/api/record/EntryProperty.java]: Hi @undx 
have them all, but i see we public something in tck document : "talend.studio.type", we need do migration on it? if that, advice to keep the old value for STUDIO_TYPE.

Comment (undx) [reply component-api/src/main/java/org/talend/sdk/component/api/record/EntryProperty.java]: `talend.studio.type` is distinct from `field.type`, it represents a studio type not a db type (ie meaning `id_String`, `id_Long`, etc.). But need to be taken in account anyway.

**Review by undx — DISMISSED** [review]

**Pull request merged**